protocol/urls/urls.go

```diff
@@ -123,35 +136,84 @@ func GetGithubPreviewData(link string) (previewData LinkPreviewData, err error)
 	return previewData, nil
 }
 
-func GetLinkPreviewData(link string) (previewData LinkPreviewData, err error) {
+func GetGiphyOembed(url string) (data GiphyOembedData, err error) {
+	oembedLink := fmt.Sprintf(GiphyOembedLink, url)
+
+	jsonBytes, err := GetURLContent(oembedLink)
 
-	url, err := url.Parse(link)
 	if err != nil {
-		return previewData, fmt.Errorf("Cant't parse link %s", link)
+		return data, fmt.Errorf("Can't get bytes from Giphy oembed response at %s", oembedLink)
```

Comment:
This is an example, to be clear the convention applies to error text not the error variable.